website/src/views/planner/ModuleMenu.tsx

```diff
@@ -17,11 +17,26 @@ type MenuItem = {
 };
 
 const ModuleMenu = memo((props: Props) => {
+  const [isMenuOverflowing, setMenuOverflowing] = useState<boolean>(false);
+  const [hasRendered, setHasRendered] = useState<boolean>(false);
```

Comment:
```suggestion
  const [updateState, forceUpdate] = useReducer((x) => x + 1, 0);
```

This is a slightly different way of forcing renders. The current `toggleRender` function could lead to stale state as `hasRendered` may not be consistent with the actual state during re-renders.
Alternatively, you could change `toggleRender` to use a function instead:
```js
const toggleRender = () => setHasRendered((hasRendered) => !hasRendered);
```

Reply:
Hi @li-kai , thank you for the comment. 

I have fixed the above to rectify. I have chosen the alternative method which is to change `toggleRender` to use a function instead, thank you!